server/src/internalClusterTest/java/org/elasticsearch/search/fetch/FetchSubPhasePluginIT.java

```diff
@@ -139,19 +135,18 @@ private void hitExecute(SearchContext context, HitContext hitContext) {
                 hitField = new DocumentField(NAME, new ArrayList<>(1));
                 hitContext.hit().setDocumentField(NAME, hitField);
             }
-            TermVectorsRequest termVectorsRequest = new TermVectorsRequest(context.indexShard().shardId().getIndex().getName(),
-                    hitContext.hit().getId());
-            TermVectorsResponse termVector = TermVectorsService.getTermVectors(context.indexShard(), termVectorsRequest);
-            try {
+            Terms terms = hitContext.reader().getTermVector(hitContext.docId(), field);
```

Comment:
Checking I understand, is this just an optional clean-up ?

Reply:
Right, it means that we don't need to expose the whole index shard to the FetchContext.